mergify_engine/queue/merge_train.py

```diff
@@ -806,20 +830,44 @@ async def update_state(
         self.last_evaluated_conditions = evaluated_queue_rule.conditions.get_summary()
         self.last_checks = []
         self.has_timed_out = False
+        self.ci_has_passed = False
+
+        ci_conditions_list = [
+            condition.match
+            for condition in evaluated_queue_rule.conditions.walk()
+            if condition.get_attribute_name().startswith("check-")
+            or condition.get_attribute_name().startswith("status-")
+        ]
+        if all(ci_conditions_list):
+            self.ci_has_passed = True
+
+        rules = await self.train.get_queue_rules()
+        timeout = (
+            (
+                rules[self.initial_embarked_pulls[0].config["name"]].config[
+                    "checks_timeout"
+                ]
+                if "checks_timeout"
```

Comment:
On parsed configuration all keys are always present (but maybe set to None if voluptuous allow it).